app/components/permitted-stacks-by-role/component.js

```diff
@@ -0,0 +1,21 @@
+import Ember from 'ember';
+
+export default Ember.Component.extend({
+  permittedStacks: function() {
+    return this.get('stacks').filter((stack) => {
```

Comment:
`get` is very cheap, but not free. Additionally, `get` will return undefined if anything in the `complex.path` passed to it is not there.

Additionally we've got a method that handles this logic for you! I suggest:

```javascript
  updatePermittedStacks: function() {
    const role = this.get('role');
    const scope = this.get('scope');
    Ember.RSVP.all(this.get('stacks').map((stack) = > {
      return stack.permitsRole(role, scope).then((permitted) => {
        return permitted && stack;
      });
    })).then((stacks) => {
      const permittedStacks = stacks.filter(r => !!r);
      this.set('permittedStacks', permittedStacks);
    });
  }.observes('scope', 'stacks.[]', 'role.privileged')
```